initrd/etc/luks-functions

```diff
@@ -295,153 +309,267 @@ select_luks_container()
+                read -r luks_new_Disk_Recovery_Key_passphrase
+            done
+
+            echo -e "\nEnter the current LUKS Disk Recovery Key passphrase (Configured at OS installation or by OEM):"
+            read -r luks_current_Disk_Recovery_Key_passphrase
+        fi
+
+        echo -n "$luks_current_Disk_Recovery_Key_passphrase" > /tmp/secret/luks_current_Disk_Recovery_Key_passphrase
+        echo -n "$luks_new_Disk_Recovery_Key_passphrase" > /tmp/secret/luks_new_Disk_Recovery_Key_passphrase
+
+        DEBUG "$luks_container: Test unlocking of LUKS encrypted drive content with current LUKS Disk Recovery Key passphrase..."
+        if ! DO_WITH_DEBUG cryptsetup open --test-passphrase "$luks_container" --key-file /tmp/secret/luks_current_Disk_Recovery_Key_passphrase > /dev/null 2>&1; then
+            whiptail --title "$luks_container: Wrong current LUKS Disk Recovery Key passphrase?" --msgbox \
+                "If you previously changed it and do not remember it, you will have to\n reinstall the OS from an external drive.\n\nTo do so, place the ISO file and its signature file on root of an\n external drive, and select Options-> Boot from USB \n\nHit Enter to retry." 0 80
+            TRACE_FUNC
+            detect_boot_device
+            mount -o remount,rw /boot
+            rm -f /boot/kexec_key_devices.txt
+            mount -o remount,ro /boot
+            luks_secrets_cleanup
+            unset LUKS
+            continue
+        fi
+
+        echo -e "\nChanging $luks_container LUKS encrypted disk passphrase to the new LUKS Disk Recovery Key passphrase..."
+        if ! DO_WITH_DEBUG cryptsetup luksChangeKey "$luks_container" --key-file=/tmp/secret/luks_current_Disk_Recovery_Key_passphrase /tmp/secret/luks_new_Disk_Recovery_Key_passphrase; then
+            whiptail --title 'Failed to change LUKS passphrase' --msgbox \
+                "Failed to change the passphrase for $luks_container.\nPlease try again." 0 80
+            continue
+        fi
+
+        echo "Success changing passphrase for $luks_container."
+    done
+
+    # Export the new passphrase if all containers were processed successfully
+    luks_current_Disk_Recovery_Key_passphrase=$luks_new_Disk_Recovery_Key_passphrase
+    export luks_current_Disk_Recovery_Key_passphrase
+    export luks_new_Disk_Recovery_Key_passphrase
+    export LUKS
 }
 
 luks_secrets_cleanup()
 {
+	TRACE_FUNC
+
 	#Cleanup
-	shred -n 10 -z -u /tmp/luks_new_Disk_Recovery_Key_passphrase 2>/dev/null || true
-	shred -n 10 -z -u /tmp/luks_current_Disk_Recovery_Key_passphrase 2>/dev/null || true
+	shred -n 10 -z -u /tmp/secret/luks_new_Disk_Recovery_Key_passphrase 2>/dev/null || true
+	shred -n 10 -z -u /tmp/secret/luks_current_Disk_Recovery_Key_passphrase 2>/dev/null || true
+
+	#Unset variables (when in same boot)
 	unset luks_current_Disk_Recovery_Key_passphrase
 	unset luks_new_Disk_Recovery_Key_passphrase
+
+	#TODO: refactor logic of selec_luks_conatainer, where to put	
+	#unset LUKS
```

Comment:
unsetting $LUKS should be done sowhere more globally, but DOD: where?

Reply:
I have no idea where.  Like I said, it's really hard to follow this LUKS variable around to figure out what's supposed to happen in lots of corner cases.

Like I said here (https://github.com/linuxboot/heads/pull/1541#discussion_r1735119761), it would be much easier to understand if the function did not do anything with a global variable.

It'd be some refactoring, but IMO to make this understandable:
- have the function always do what it says it will do (prompt), never decide that we're not really going to prompt
- don't produce output in a global variable, produce output on stdout for the caller to capture
- let the callers manage this state locally and decide whether to reuse the existing value or prompt